### ISSUE TITLE
RTK Query Possibly Bug: BigInt type is being rounded in response 

### ISSUE THREAD

**Issue by yasen002** [reduxjs/redux-toolkit#3330]
Hi, 
I was calling an RTK query, and found a behavior that might be a bug. 

I created a GET query to get some customer data, and discovered that the last digit of customer ID (18 digit BigInt) is being rounded. 

I thought it might be an API issue so I created a mock API, and got the same result.

I tried using  Fetch API, and it returned the correct data.

**sample data**:
`
{
"customer":{
"name":"Jason", 
"id":123456789012345678
}
}
`

**RTK Query response data**
`
{
"customer":{
"name":"Jason", 
"id":123456789012345670
}
}
`

**version**:  "@reduxjs/toolkit": "^1.8.6"

**reproduce** : 
create an RTK query GET method to call an API that returns an 18 digit long number in the response body

**expected behavior**
expected to return a response data with 18 digit long number in the response body

**current behavior **
Rounding the last digit of the long number to '0'

**Comment by phryneas** [comment]
That's the normal behaviour of JavaScript's
```js
JSON.parse('{ "customer":{ "name":"Jason", "id":123456789012345670 } }')
```
which results in 
```json
{
  "customer": {
    "name": "Jason",
    "id": 123456789012345660
  }
}
```

You will need to implement a custom `responseHandler` for this.

**Comment by yasen002** [comment]
@phryneas Thank you so much!

**Comment by yasen002** [comment]
just for anyone faces this issue in the future, here is how I solved it:
`import JSONbig from "json-bigint";

/**
 * This is a custom response handler, and the pur pose of this handler is to
 * use JSONbig library to capture big integer into JSON.
 * customerId and sequence numbers are coming in as 18 digit big int, and
 * when we parse the response using JSON.parse method, last few digits of those
 * big ints are being rounded to 0. With this JSONbig library, we were able to
 * parse full 18 digit IDs.
 * @param response
 * @returns promise
 */
export const customResponseHandler = (response: Response): Promise<any> => {
  const JSONbigString = JSONbig({ storeAsString: true });

  return response.text().then((res) => {
    return JSONbigString.parse(res);
  });
};
`
